frontend/src/components/phones/onboarding/RelayNumberPicker.tsx

```diff
@@ -103,35 +95,26 @@ const RelayNumberSelection = (props: RelayNumberSelectionProps) => {
   const [relayNumberIndex, setRelayNumberIndex] = useState(0);
 
   useEffect(() => {
-    _getRelayNumberSuggestions();
-  }, []);
-
-  // GET request to get relay number suggestions
-  const _getRelayNumberSuggestions = () => {
-    setIsLoading(true);
-    getRelayNumberSuggestions()
-      .then((response) => response.json())
-      .then((data) => {
-        const { other_areas_options, same_area_options, same_prefix_options } =
-          data;
-        const suggestedNumbers: RelayNumberOption[] = [
-          ...same_area_options,
-          ...same_prefix_options,
-          ...other_areas_options,
-        ];
+    if (relayNumberSuggestionsData.data) {
+      // get all relay number suggestions from our API
+      const { other_areas_options, same_area_options, same_prefix_options } =
+        relayNumberSuggestionsData.data;
+      // combine all relay number suggestions into one array
+      const suggestedNumbers = [
+        ...same_area_options,
+        ...other_areas_options,
+        ...same_prefix_options,
+      ];
 
-        setRelayNumberSuggestions(
-          suggestedNumbers.map(
-            (suggestion: RelayNumberOption) => suggestion.phone_number
-          )
-        );
+      // set the relay number suggestions - an array of strings
+      setRelayNumberSuggestions(
+        suggestedNumbers.map((suggestion) => suggestion.phone_number)
+      );
 
-        setIsLoading(false);
-      })
-      .catch((error) => {
-        // Do nothing, just keep the default values
-      });
-  };
+      // remove the loading spinner
+      setIsLoading(false);
```

Comment:
Additionally, the `isLoading` state variable is no longer needed either - just replace every instance of `isLoading` with `!relayNumberSuggestionsData.data`.

An additional advantage is that if you do that in an `if`, TypeScript can infer that `.data` is _not_ `undefined` in the `else` branch (or after a `return` in that `if`).

Reply:
Done - yes, this is much cleaner. I think I was getting a react minification error 301/321 at some point, but I think it was my placement - I like this a lot better.